src/Workspaces/Core/Portable/Serialization/SerializerService.cs

```diff
@@ -84,7 +84,7 @@ public Checksum CreateChecksum(object value, CancellationToken cancellationToken
                     return CreateChecksum((AnalyzerReference)value, cancellationToken);
 
                 case WellKnownSynchronizationKind.SerializableSourceText:
-                    return Checksum.Create(((SerializableSourceText)value).ContentHash);
+                    throw new InvalidOperationException("Clients can already get a checksum directly from a SerializableSourceText");
```

Comment:
i honestly want to delete this entire method (which just takes in object).  All callers know what types they are serializing,  so they should be able to call strongly typed helpers.  However, there's some more work to be done in the test code to make this simpler.  so it will be in a followup.